### PR TITLE
feat: ✨ Generate proof for multitple file chunks

### PR DESCRIPTION
**Pull request by snowmead**

### CODE REVIEW

File: node/src/tasks/bsp_upload_file_task.rs [hunk 1]
```diff
@@ -100,22 +100,25 @@ where
         let file_key: HasherOutT<SHC::TrieLayout> = TryFrom::try_from(*event.file_key.as_ref())
             .map_err(|_| anyhow::anyhow!("File key and HasherOutT mismatch!"))?;
 
+        let proven = event
+            .chunk_with_proof
+            .proven
+            .first()
+            .ok_or_else(|| anyhow::anyhow!("Proven should contain one element; qed"))?;
```
Comment (links234) [inline node/src/tasks/bsp_upload_file_task.rs:107]: This could be multiple chunks too, maybe even required later for better upload speeds.

**Review by links234 — APPROVED** [review]
LGTM!

File: node/src/tasks/bsp_upload_file_task.rs (hunk 1 above)
Comment (ffarall) [reply node/src/tasks/bsp_upload_file_task.rs:107]: Agreed, let's add a TODO here.

**Review by ffarall — APPROVED** [review]
Make sure to solve conflicts with main, and then ready to merge. Concise and great work.

**Pull request merged**